packages/apps/human-app/frontend/src/modules/worker/profile/components/wallet-connection-control.tsx

```diff
@@ -0,0 +1,51 @@
+import { useTranslation } from 'react-i18next';
+import { Button } from '@/shared/components/ui/button';
+import { type WorkerProfileStatus } from '../types';
+import { DoneLabel } from './status-labels';
+import { WalletConnectDone } from './wallet-connect-done';
+
+interface WalletConnectionControlProps {
+  status: Pick<WorkerProfileStatus, 'kycApproved'>;
+  isConnected: boolean;
+  hasWalletAddress: boolean;
+  isRegisterAddressPending: boolean;
+  onConnect: () => void;
+}
+
+export function WalletConnectionControl({
+  status,
+  isConnected,
+  hasWalletAddress,
+  isRegisterAddressPending,
+  onConnect,
+}: Readonly<WalletConnectionControlProps>) {
+  const { t } = useTranslation();
+
+  if (!status.kycApproved) {
+    return (
```

Comment:
Maybe we can remove it in favor of the button return at the very end of this component?